frontend/src/leaderboard/Leaderboard.tsx

```diff
@@ -1,58 +1,98 @@
+import React, { FC, useEffect, useState } from "react";
 import { AxiosError } from "axios";
-import { getCurrentSeason, getPlayerLeaderboard } from "../api/LeaderboardAPI";
+import { getPlayerLeaderboard } from "../api/LeaderboardAPI";
 import { getGameTypeString } from "../common/Utils";
 import { Container, Table } from "react-bootstrap";
+import Select from "react-select";
+import { getSeasonsAPI } from "../api/AdminAPI";
 
 const Leaderboard: FC<GameTypeProp> = ({ gameVariant }) => {
-    const [currentSeason, setCurrentSeason] = useState<Season | undefined>(undefined);
+    const [seasons, setSeasons] = useState<Season[]>([]);
+    const [selectSeason, setSelectSeason] = useState<Season | undefined>(undefined);
     const [leaderboard, setLeaderboard] = useState<LeaderboardType[]>([]);
-    const [noSeasonsMessage, setNoSeasonsMessage] = useState<string>("");
+    const [loading, setLoading] = useState(true);
 
     useEffect(() => {
-        getPlayerLeaderboard(gameVariant)
-            .then((response) => {
-                const playerElos: LeaderboardType[] = response.data.players.sort((a, b) => {
-                    return Number(b.elo) - Number(a.elo);
-                });
-                const oneDecimalEloLeaderboard: LeaderboardType[] = playerElos.map((player) => {
-                    const elo = Number(player.elo);
+        if (typeof selectSeason !== "undefined") {
+            getPlayerLeaderboard(gameVariant, selectSeason.id)
+                .then((response) => {
+                    const playerElos: LeaderboardType[] = response.data.players.sort((a, b) => {
+                        return Number(b.elo) - Number(a.elo);
+                    });
+                    const oneDecimalEloLeaderboard: LeaderboardType[] = playerElos.map((player) => {
+                        const elo = Number(player.elo);
 
-                    return {
-                        ...player,
-                        elo: elo.toFixed(1),
-                    };
+                        return {
+                            ...player,
+                            elo: elo.toFixed(1),
+                        };
+                    });
+                    setLeaderboard(oneDecimalEloLeaderboard);
+                })
+                .catch((error: AxiosError) => {
+                    console.log("Error fetching leaderboard: ", error.response?.data);
                 });
-                setLeaderboard(oneDecimalEloLeaderboard);
-            })
-            .catch((error: AxiosError) => {
-                console.log("Error fetching leaderboard: ", error.response?.data);
-            });
-    }, [gameVariant]);
+        }
+    }, [gameVariant, selectSeason]);
 
     useEffect(() => {
-        getCurrentSeason()
+        getSeasonsAPI()
             .then((response) => {
-                setCurrentSeason(response.data);
+                const allSeasons = response.data.pastSeasons;
+                if (response.data.currentSeason) {
```

Comment:
wouldn't it be preferable to just include currentSeason directly into the all seasons list?